### DISCUSSION TITLE
[Feature]: Shared database + service extension + database backup

### DISCUSSION THREAD

**Discussion by jugglingjsons** [coollabsio/coolify#3400] [Feature Requests]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

It could be useful to define a database as a shared one and let multiple service use the same database server.
Services could be extended with a "use shared db toggle" if a shared database of a kind was active and have a mapping for the db credentials / envs required configured in prisma.

### Why should this be worked on?

It would potentially:
- limit the resource usage
- make it a little easier to centralise backups for databases

Sample flow:
```mermaid
graph LR
    servuceCreation-->|consultCoolifyState|ifRequiredDatabaseExists-->|displayToggle|useSharedDB-->|mapDBConfig|toRequiredEnvSettings
```
and for backup
```mermaid
graph LR
    sharedDB-->|backup|localFileDump-->|watchFile|triggerOnChange-->|replicate|s3
```

**Comment by andrasbacsai** [comment]
This will be implemented in v4 as soon as services are added.

I'm closing this issue, because tracking features will be moved to GitHub discussions. Github issues will be only for bugs.

Version 4 is in full focus. v3 related issues will be closed, if it is not a critical bug.

Thanks for your understanding.

**Comment by mahamudml** [comment]
Is this implemented in v4, was waiting for this feature

**Comment by sneko** [comment]
In the meantime if I use a shared database in a dedicated team, can another resource uses it from another team? Or is each team scoped to its own docker network? If so, how can I make it accessible with an internal address cross-teams?

(just asking, I will test soon if it's possible)

**Comment by sovetski** [comment]
any news here?
